test/automation/src/code.ts

```diff
@@ -201,17 +201,23 @@ export class Code {
 				while (!done) {
 					retries++;
 
-					if (retries > 20) {
-						this.logger.log('Smoke test exit call did not terminate process after 10s, still trying...');
+					if (retries === 20) {
+						this.logger.log('Smoke test exit call did not terminate process after 10s, forcefully exiting the application...');
+
+						try {
+							process.kill(this.mainProcess.pid);
```

Comment:
I like the intent, but should this be `tree-kill`, similar to: https://github.com/microsoft/vscode/blob/99fabbdb270ee1b22e96ded3f85ae308962cff33/test/automation/src/electronDriver.ts#L139

Reply:
And I also think that any exception from that call should be logged as it could hint to the underlying issue.